src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -48,23 +57,39 @@ public RunReadyToRunCompiler()
 
         protected override bool ValidateParameters()
         {
-            _crossgenPath = CrossgenTool.ItemSpec;
-            _clrjitPath = CrossgenTool.GetMetadata("JitPath");
-            _diasymreaderPath = CrossgenTool.GetMetadata("DiaSymReader");
+            _createPDBCommand = CompilationEntry.GetMetadata("CreatePDBCommand");
 
-            if (!File.Exists(_crossgenPath) || !File.Exists(_clrjitPath))
+            if (CrossgenTool == null && Crossgen2Tool == null)
             {
                 return false;
```

Comment:
Already like this, so not high pri, but what is the error when this returns false. Should we consider logging the individual cases? I suppose these are mainly thing that are not expected to actually happen?

Reply:
Correct, this is not expected to actually happen. If we fail to resolve the crossgen tool earlier in the pipeline (PrepareForReadyToRunCompilation), the task will show meaningful messages.